docs/troubleshooting-rbe.md

```diff
@@ -37,3 +37,35 @@ If you'd like to add Mac executors to your BuildBuddy Cloud account, please [con
 This error occurs when your build is configured for darwin (Mac OSX) CPUs, but attempting to run on Linux executors. Mac executors are not included in BuildBuddy Cloud's free-tier offering.
 
 If you'd like to add Mac executors to your BuildBuddy Cloud account, please [contact our sales team](/request-demo/).
+
+## WARNING: Remote Cache: UNAVAILABLE: io exception
+
+This error occurs when Bazel JVM have a hard time maintaining the TCP connection with BuildBuddy remote cache.
+
+To validate this error, user could use Bazel's flag `--remote_grpc_log=grpc.log` to capture the grpc traffic
+from Bazel JVM and BuildBuddy. The log file will be in protobuf format. to convert it to JSON format, user
+could use our [BB CLI](/docs/cli) and run `bb print --grpc_log=<path-to-file>/grpc.log`. In side the log, we
+expect to see network errors such as
+
+```json
+  "status":  {
+    "code":  14,
+    "message":  "io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer"
+  },
+```
+
+This is usually encountered by customers who run operate some proxy or gateway (i.e. AWS NAT Gateway) in between
+Bazel JVM and BuildBuddy server.
+
+When this happen, user should consider tunning their Linux network settings like this:
+
+```bash
+sudo sysctl -w net.ipv4.tcp_keepalive_time=180
+sudo sysctl -w net.ipv4.tcp_keepalive_intvl=60
+sudo sysctl -w net.ipv4.tcp_keepalive_probes=25
```

Comment:
This value seems unnecessarily high. This is effectively saying don't consider the connection dead until 180+60*25=28 minutes after sending the first keepalive packet. I would aim for something closer to 5-10 minutes.

Reply:
Here is the kernel default https://github.com/torvalds/linux/blob/7ba2090ca64ea1aa435744884124387db1fac70f/include/net/tcp.h#L155-L157

which is
```
7200
75
9
```

which total to roughly more than 2 hours. Another point to note is that on the Client machine, anything could be running, not just Bazel. So we probably want to stick with the default as much as possible.

I think the goal here is not to extend the total keepalive time but to shorten the ping interval to bypass the proxy/gateway limit in the middle. So how about we use the default value for `tcp_keepalive_intvl` and `tcp_keepalive_probes`, but still mention them in the doc for users' awareness? That would make it 

```bash
sudo sysctl -w net.ipv4.tcp_keepalive_time=180
# Using default value
sudo sysctl -w net.ipv4.tcp_keepalive_intvl=75
# Using default value
sudo sysctl -w net.ipv4.tcp_keepalive_probes=9
```

which equate to 180 + 75 * 9 > 14 mins.

Reply:
The kernel defaults were set 20+ years ago when connectivity was very different. It would be very unusual on modern day networks and modern day PCs to not get a response to a keepalive ping in 5 minutes, let alone 15.

Reply:
@vadimberezniker Alright. Let's do `180 + 60 * 2 = 300s = 5m` then.